### ISSUE TITLE
a11y: fix semantics for MostViewedFooterGrid tabs

### ISSUE THREAD

**Issue by bryophyta** [guardian/dotcom-rendering#6311]
> **Note**
> Some [work is planned](https://github.com/guardian/dotcom-rendering/issues/6247) for the data fetching part of 'Most Viewed'. Given that this might affect the data model and API that we use for rendering the container, it might make sense to address these accessibility issues after the data work has been done.

There are a couple of related issues in the existing markup for the Most Viewed links container in both Frontend and DCR:

## 1. List items not wrapped in list container

One of the remaining accessibility issues for Fronts being reported by Lighthouse is that:

```List items (<li>) are not contained within <ul> or <ol> parent elements.```

The highlighted elements are the card links in the Most Viewed container:

<img width="980" alt="image" src="https://user-images.githubusercontent.com/37048459/198679313-e51c3666-308e-4e29-ab7d-433b7bba728a.png">

The `<li>`s are in fact contained in an `<ol>`, so I think that Lighthouse is responding to the fact that the `<ol>` has `role=tabpanel` set,  which overrides the default role of `list`.

A quick fix for the issue flagged by Lighthouse would be to wrap the `<ol>` (e.g. in a `<section>`) and put the `tabpanel` role on the wrapper instead.

## 2. Container still marked up as tablist when there's only one 'tab'

In the case where there's only one tab (e.g. on network Fronts) the aria roles are still set up to describe the layout as a list of tabs and tab panels. This is misleading in the case where there are no 'other' tabs to navigate between.

A fix here might be to check whether there is more than one set of links. If there is more than one, render the tabbed layout. If there's only one, render a simpler version of the layout, without the aria-roles.

## 3. Keyboard navigation

**Update:** I'm pretty sure that this should be considered as a separate issue, if it is an issue. As @isabellapurchas points out below, we currently only have two tabs at most, so having to tab (key) through both tabs (links) shouldn't be too much of a UX problem.

[Heydon Pickering suggests](https://inclusive-components.design/tabbed-interfaces/) adapting the keyboard behaviour for tabbed interfaces using JavaScript. Their reasoning seems good, although I don't know whether the suggested pattern would be familiar to users. Getting the keyboard behaviour right seems like a separate issue to me, though.

## Resources

- [Tabbed Interfaces](https://inclusive-components.design/tabbed-interfaces/) in Heydon Pickering's _Inclusive Components_ series.
- [W3c: Example of Tabs with Manual Activation](https://www.w3.org/WAI/ARIA/apg/example-index/tabs/tabs-manual.html)
- [gov.uk design system tabs component](https://design-system.service.gov.uk/components/tabs/)
- [gov.uk github discussion on tabs](https://github.com/alphagov/govuk-design-system-backlog/issues/100)
- These issues were also picked up in two DAC audit reports: 
   1. [Report on the list semantics](https://drive.google.com/file/d/1vp5M4hxnNBZkh3jronwR8xorzvylb584/view?usp=sharing).
   2. [Report on the discoverability of the tabs](https://drive.google.com/file/d/1OVYBrDL0-xQG6IxocZTJzY-gTJ96iRzT/view)

**Comment by bryophyta** [comment]
@isabellapurchas this probably won't be worked on for at least a couple of weeks, but it would be good to hear your  thoughts on this if you have the time?

**Comment by isabellapurchas** [comment]
@bryophyta Thanks for adding all the information for this. I agree the functionality with the tab panels needs fixing, and like the idea of checking how many lists need to be rendered and not rendering a tabbed layout if there is only one. The different keyboard behaviour from Heydon Pickering (https://inclusive-components.design/tabbed-interfaces/) is interesting, it makes sense but I also don't know if the pattern would be familiar to users. It seems like this would be useful if we had more than 2 tabs which I don't believe we do (is that correct?).
Something else to check is whether the 'most viewed' context is communicated to the user when they are in those tab lists. If not, perhaps we can include a label to explain what those tabs are for.

Is there anything else you need UX input on for this?

**Comment by bryophyta** [comment]
putting these as a note-to-self; planning to come back and write up notes based on them:

- [gov.uk design system tabs component](https://design-system.service.gov.uk/components/tabs/)
- [gov.uk github discussion on tabs](https://github.com/alphagov/govuk-design-system-backlog/issues/100)
